File: typescript/tests/integration/beta-skills-metadata.integration.test.ts
/**
 * Integration tests for BetaSkills.listMetadata().
 *
 * Note: This test calls the real backend. Do not run concurrently.
 */

import { AgentBay } from "../../src/agent-bay";
import { log } from "../../src/utils/logger";

describe("Beta skills metadata integration tests", () => {
  const apiKey = process.env.AGENTBAY_API_KEY;
  let agentBay: AgentBay;

  beforeEach(() => {
    if (apiKey) {
      agentBay = new AgentBay({ apiKey });
    }
  });

  test("should list official skills metadata", async () => {
    if (!apiKey) {
      log("Skipping test: AGENTBAY_API_KEY not set");
      return;
    }

    const items = await agentBay.betaSkills.listMetadata();
    expect(Array.isArray(items)).toBe(true);
    expect(items.length).toBeGreaterThan(0);

    const first = items[0];
    expect(typeof first.name).toBe("string");
    expect(first.name.trim().length).toBeGreaterThan(0);
    expect(typeof first.description).toBe("string");

    expect((first as any).dir).toBeUndefined();
  }, 30000);
});

File: golang/tests/pkg/integration/beta_skills_metadata_integration_test.go
package integration

import (
	"os"
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
	"github.com/stretchr/testify/require"
)

func TestBetaSkillsListMetadata(t *testing.T) {
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Skip("Skipping integration test: AGENTBAY_API_KEY not set")
	}

	ab, err := agentbay.NewAgentBay(apiKey)
	require.NoError(t, err)

	items, err := ab.BetaSkills.ListMetadata()
	require.NoError(t, err)
	require.Greater(t, len(items), 0)
	require.NotEmpty(t, items[0].Name)
}
